lib/ecto/migration/schema_migration.ex

```diff
@@ -38,4 +28,42 @@ defmodule Ecto.Migration.SchemaMigration do
   def down(repo, version) do
     repo.delete_all from(p in __MODULE__, where: p.version == ^version), @opts
   end
+
+  defp create_migrations_table(Ecto.Adapters.MySQL = adapter, repo) do
```

Comment:
Why couldn't you use `CREATE TABLE IF NOT EXISTS` here? We should never special case behaviour for adapters.